eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java

```diff
@@ -31,63 +31,82 @@
 import java.util.List;
 import java.util.Map;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class EventMeshRecommendImpl implements EventMeshRecommendStrategy {
-    
-    private final EventMeshTCPServer eventMeshTCPServer;
 
-    public EventMeshRecommendImpl(EventMeshTCPServer eventMeshTCPServer) {
+    private static final int DEFAULT_PROXY_NUM = 1;
+
+    private final transient EventMeshTCPServer eventMeshTCPServer;
+
+    public EventMeshRecommendImpl(final EventMeshTCPServer eventMeshTCPServer) {
         this.eventMeshTCPServer = eventMeshTCPServer;
     }
 
     @Override
-    public String calculateRecommendEventMesh(String group, String purpose) {
+    public String calculateRecommendEventMesh(final String group, final String purpose) throws Exception {
         List<EventMeshDataInfo> eventMeshDataInfoList;
-        if (StringUtils.isAnyBlank(group, purpose)) {
-            log.warn("EventMeshRecommend failed,params illegal,group:{},purpose:{}", group, purpose);
+
+        if (StringUtils.isBlank(group) || StringUtils.isBlank(purpose)) {
+            if (log.isWarnEnabled()) {
+                log.warn("EventMeshRecommend failed,params illegal,group:{},purpose:{}", group, purpose);
+            }
             return null;
         }
+
         final String cluster = eventMeshTCPServer.getEventMeshTCPConfiguration().getEventMeshCluster();
         try {
             eventMeshDataInfoList = eventMeshTCPServer.getRegistry().findEventMeshInfoByCluster(cluster);
         } catch (Exception e) {
-            log.warn("EventMeshRecommend failed, findEventMeshInfoByCluster failed, cluster:{}, group:{}, purpose:{}, errMsg:{}",
-                    cluster, group, purpose, e);
+            if (log.isWarnEnabled()) {
+                log.warn("EventMeshRecommend failed, findEventMeshInfoByCluster failed, cluster:{}, group:{}, purpose:{}, errMsg:{}",
+                        cluster, group, purpose, e);
+            }
             return null;
         }
 
-        if (CollectionUtils.isEmpty(eventMeshDataInfoList)) {
-            log.warn("EventMeshRecommend failed,not find eventMesh instances from registry,cluster:{},group:{},purpose:{}",
-                    cluster, group, purpose);
+        if (eventMeshDataInfoList == null || CollectionUtils.isEmpty(eventMeshDataInfoList)) {
```

Comment:
<img width="518" alt="image" src="https://user-images.githubusercontent.com/63457588/210812394-6faf1795-881f-4d07-8cc8-f68a27c76641.png">

`CollectionUtils.isEmpty` has involved the null pointer check

Reply:
I have fixed it. @horoc